packages/store-sync/src/world/getSystemAbi.ts

```diff
@@ -0,0 +1,35 @@
+import { Client, Abi, Address, hexToString, parseAbiItem } from "viem";
+import metadataConfig from "@latticexyz/world-module-metadata/mud.config";
+import { getRecords } from "../getRecords";
+
+export async function getSystemAbi({
+  client,
+  worldAddress,
+  fromBlock,
+  toBlock,
+  indexerUrl,
+  chainId,
+}: {
+  readonly client: Client;
+  readonly worldAddress: Address;
+  readonly fromBlock?: bigint;
+  readonly toBlock?: bigint;
+  readonly indexerUrl?: string;
+  readonly chainId?: number;
```

Comment:
do we want system ID as a param?

Reply:
what do you mean by "abi + system ID for each system ABI"? Kind of get it but not sure how this would look exactly. Anyhow, I'd be more inclined towards this returning the full system ABI so you don't have to fetch each one individually.

Reply:
you're generally only interacting with one or a handful of systems, not the entire world's systems, esp as worlds get bigger, so I am more inclined to do individual ones (i.e. system ID param)

you always need a system ID in addition to the ABI + func name to call the system

Reply:
Yeah actually makes sense when put that way. Wonder if we should have another helper for querying the list of systems in that case?

Reply:
and I'll add `getSystemAbis()` as well since querying multiple systems could be PITA.

Reply:
> Yeah actually makes sense when put that way. Wonder if we should have another helper for querying the list of systems in that case?

You already can by reading the world's system tables